lib/axiosHttpClient.ts

```diff
@@ -103,6 +103,12 @@ export class AxiosHttpClient implements HttpClient {
       bodyType === "function" ? httpRequest.body() :
       httpRequest.body;
 
+    const userUploadProgress = httpRequest.onUploadProgress;
+    const onUploadProgress = userUploadProgress && ((rawEvent: ProgressEvent) => userUploadProgress({ loaded: rawEvent.loaded, total: rawEvent.lengthComputable ? rawEvent.total : undefined }));
+
+    const userDownloadProgress = httpRequest.onDownloadProgress;
+    const onDownloadProgress = userDownloadProgress && ((rawEvent: ProgressEvent) => userDownloadProgress({ loaded: rawEvent.loaded, total: rawEvent.lengthComputable ? rawEvent.total : undefined }));
```

Comment:
Can you add type definitions for these variables? Adding the raw type could be bad, so I'd recommend adding typedefs at the top of this file (UploadProgressFunction and DownloadProgressFunction) and then using them here.

Reply:
Do you mean like `const onUploadProgress: ProgressFunction = ...`?

Reply:
Yes.